pallets/asset/src/lib.rs

```diff
@@ -2176,6 +2194,37 @@ impl<T: Trait> Module<T> {
         (transfer_status, weight_for_valid_transfer)
     }
 
+    /// Sets the initial total supply of a confidential asset.
+    /// Only called by the token owner of a confidential asset. Can be called only once.
+    ///
+    /// # Arguments
+    /// * `origin` Secondary key of the token owner.
+    /// * `ticker` Ticker of the token.
+    /// * `total_supply` Ticker of the token.
+    pub fn set_total_supply(
+        did: IdentityId,
+        ticker: Ticker,
+        total_supply: T::Balance,
+    ) -> DispatchResult {
+        ensure!(Self::is_owner(&ticker, did), Error::<T>::Unauthorized);
+        // Read the token details
+        let mut token = Self::token_details(&ticker);
+        // TODO: either change SecurityToken, or add a new Storage
+        //ensure!(token.is_confidential, Error::<T>::NonConfidentialAssetsCannotSetTotalSupply);
```

Comment:
Another option is to get it from the confidential asset pallet, but that will cause a circular dependency.

Reply:
You can create a trait that has the `is_confidential` function and implement it for the confidential module. Using that in the asset module will bypass the circular dependency. For example, see how the identity pallet uses traits like `MultiSigSubTrait` and `PortfolioSubTrait`.